Automattic/wp-calypso
# Framework: Proposed Redux page state module

This pull request seeks to add a new `state.ui.page` subtree to the global Redux state tree. The intention of this is to act as a key-value mapping with the intended lifetime of a single page view. A universal route handler resets the state on each route change.

This contradicts some [earlier statements I had made](https://github.com/Automattic/wp-calypso/pull/1378#issuecomment-163683205) in respond to @rralian 's question about whether we could store all UI state in the Redux global store. At the time, I had envisioned custom actions, reducers, and selectors for managing any particular piece of page state. The included proposal simplifies this by simply offering a generic object to which values can be assigned by key.

Acknowledged benefits of UI state in the Redux store to this point have included (a) being able to serialize all state for debugging purposes and (b) having a single place to keep state regardless of whether it's UI or not. I've also observed that it will promote self-sufficiency of components, as if we were to continue using React state, we might still find that ancestor components are forced to send state as props through a large hierarchy of components.

__Testing instructions:__

Page state is not currently used by any area of the application, though the dispatch to reset state on route change is included, so you should verify that no errors occur when loading any page.

Also confirm that tests pass by running `make test` from the project root directory or directly from `client/state/ui/page`.
